src/main/java/org/bihealth/mi/easybus/implementations/http/easybackend/BusEasybackend.java

```diff
@@ -0,0 +1,478 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bihealth.mi.easybus.implementations.http.easybackend;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.math.BigInteger;
+import java.util.Base64;
+import java.util.Iterator;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.bihealth.mi.easybus.Bus;
+import org.bihealth.mi.easybus.BusException;
+import org.bihealth.mi.easybus.BusMessage;
+import org.bihealth.mi.easybus.BusMessageFragment;
+import org.bihealth.mi.easybus.MessageFilter;
+import org.bihealth.mi.easybus.MessageManager;
+import org.bihealth.mi.easybus.Participant;
+import org.bihealth.mi.easybus.Scope;
+import org.bihealth.mi.easybus.implementations.http.ExecuteHTTPRequest;
+import org.bihealth.mi.easysmpc.resources.Resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import jakarta.ws.rs.client.Invocation.Builder;
+import jakarta.ws.rs.core.Response;
+
+/**
+ * Bus implementation with easybackend
+ * 
+ * @author Felix Wirth
+ *
+ */
+public class BusEasybackend extends Bus {
+    
+    /** Path to send messages */
+    private static final String   PATH_SEND_MESSAGE_PATTERN   = "api/easybackend/send/%s/%s";
+    /** Path to receive messages */
+    private static final String   PATH_GET_MESSAGES_PATTERN   = "api/easybackend/receive/%s";
+    /** Delete message */
+    private static final String   PATH_DELETE_MESSAGE_PATTERN = "api/easybackend/message/%s";
+    /** Purge all messages */
+    private static final String   PATH_PURGE_PATTERN          = "api/easybackend/message";
+    /** Logger */
+    private Logger                LOGGER                      = LogManager.getLogger(BusEasybackend.class);
+    /** Connection to backend */
+    private ConnectionEasybackend connection;
+    /** Thread */
+    private Thread                thread;
+    /** Message manager */
+    private MessageManager        messageManager;
+    /** Stop flag */
+    boolean                       stop                        = false;
```

Comment:
Why does this have package visibility? Probably not needed. If access from outside is required why not a getter/setter/isser?